### ISSUE TITLE
decommissioning ingest_v2 hangs

### ISSUE THREAD

**Issue by PSeitz** [quickwit-oss/quickwit#5068]
When trying to quit quickwit, it occasionally hangs at decommissioning ingester. Only `kill -9` works
```
^C2024-06-03T09:35:15.935Z  INFO quickwit_ingest::ingest_v2::ingester: decommissioning ingester
^C
```

I couldn't reproduce it, seems to happen randomly.

**Comment by guilload** [comment]
If there's data inflight, have you waited long enough for the next commit? It can take up to `commit_timeout_secs + ε.`

**Comment by PSeitz** [comment]
In some cases I didn't ingest data, so there shouldn't be any data inflight. I also didn't enable ingestv2 via `QW_ENABLE_INGEST_V2`.

**Comment by guilload** [comment]
Here is the bug.

Executing the REST API tests against a running node that does not have ingest V2 enabled still creates a shard and drops a few records in it. The shard does not get indexed and is not cleaned up. The next time you try to decommission the node without v2 enabled, we wait for the shard to be drained, which will never happen.

**Comment by rdettai** [comment]
> Executing the REST API tests against a running node that does not have ingest V2 enabled still creates a shard and drops a few records in it. The shard does not get indexed and is not cleaned up. The next time you try to decommission the node without v2 enabled, we wait for the shard to be drained, which will never happen.

I don´t think this is the only issue. See analysis in https://github.com/quickwit-oss/quickwit/pull/5283.

**Comment by rdettai** [comment]
In https://github.com/quickwit-oss/quickwit/pull/5283 I added 2 integration tests covering the two issues mentioned above:
- `ingest_tests::test_shutdown_control_plane_early_shutdown`
- `test_shutdown_separate_indexer`

Both pass on ingest V1 and fail if we enable ingest V2

**Comment by rdettai** [comment]
Some extra docs: https://github.com/quickwit-oss/quickwit/pull/5418
